client/packages/sprotty-client/src/features/layout/layout-commands.ts

```diff
@@ -0,0 +1,416 @@
+            bounds.newBounds.x = element.bounds.x - halfDiffWidth;
+            bounds.newBounds.width = targetWidth;
+        });
+    }
+
+    resizeHeight(elements: SelectableBoundsAware[]) {
+        const targetHeight = this.action.reductionFunction(...elements.map(element => element.bounds.height));
+        this.dispatchResizeActions(elements, (element, bounds) => {
+            // resize around middle
+            const halfDiffHeight = 0.5 * (targetHeight - element.bounds.height);
+            bounds.newBounds.y = element.bounds.y - halfDiffHeight;
+            bounds.newBounds.height = targetHeight;
+        });
+    }
+
+    resizeWidthAndHeight(elements: SelectableBoundsAware[]) {
+        const targetWidth = this.action.reductionFunction(...elements.map(element => element.bounds.width));
+        const targetHeight = this.action.reductionFunction(...elements.map(element => element.bounds.height));
+        this.dispatchResizeActions(elements, (element, bounds) => {
+            // resize around center and middle
+            const halfDiffWidth = 0.5 * (targetWidth - element.bounds.width);
+            const halfDiffHeight = 0.5 * (targetHeight - element.bounds.height);
+            bounds.newBounds.x = element.bounds.x - halfDiffWidth;
+            bounds.newBounds.y = element.bounds.y - halfDiffHeight;
+            bounds.newBounds.width = targetWidth;
+            bounds.newBounds.height = targetHeight;
+        });
+    }
+
+    dispatchResizeActions(elements: SelectableBoundsAware[], change: (element: SelectableBoundsAware, bounds: WriteableElementAndBounds) => void) {
+        const elementAndBounds: ElementAndBounds[] = []; // client- and server-side resize
+        elements.forEach(element => elementAndBounds.push(this.createElementAndBounds(element, change)));
+        this.dispatchActions([new SetBoundsAction(elementAndBounds), new ChangeBoundsOperationAction(elementAndBounds)]);
+    }
+
+    createElementAndBounds(element: SelectableBoundsAware, change: (element: SelectableBoundsAware, bounds: WriteableElementAndBounds) => void) {
+        const bounds: WriteableElementAndBounds = {
+            elementId: element.id,
+            newBounds: {
+                x: element.bounds.x,
+                y: element.bounds.y,
+                width: element.bounds.width,
+                height: element.bounds.height
+            }
+        };
+        change(element, bounds);
+        return bounds;
+    }
+
+    undo(context: CommandExecutionContext): CommandResult {
+        // we dispatch another action which can be undone, so no explicit implementation necessary
+        return context.root;
+    }
+
+    redo(context: CommandExecutionContext): CommandResult {
+        // we dispatch another action which can be redone, so no explicit implementation necessary
+        return context.root;
+    }
+}
+
+export class AlignElementsCommand extends LayoutElementsCommand {
```

Comment:
According to sprotty guidelines, commands should be annotated with `@injectable`

Reply:
Done